src/mono/wasi/include/pthread.h

```diff
@@ -0,0 +1,230 @@
+// The WASI SDK doesn't include pthread.h, but the definitions are required by various parts of the Mono sources
+// On certain runtimes such as WAMR there is actually an implementation provided for the pthread APIs so threading
+// does work. However that's not yet in the WASI standard, so we're not yet relying on being able to call the ones
+// that really start up new threads.
+
+// This file is duplicated from https://github.com/WebAssembly/wasi-libc/blob/ad5133410f66b93a2381db5b542aad5e0964db96/libc-top-half/musl/include/pthread.h
+// with small edits for compatibility. We may be able to remove it once https://github.com/WebAssembly/wasi-libc/issues/209 is resolved
+
+// This file is multi-licensed:
```

Comment:
Maybe put the license notice at the top, it's convention to have license notices at the top AFAIK

Reply:
Done